plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -456,41 +454,24 @@ const constructReviewAssignmentObject = (
 // Checks if existing assignment, should keep status and update if isLocked
 const getNewOrExistingAssignmentStatus = (
   existingReviewsAssigned: ExistingReviewAssignment[],
-  canMakeFinalDecision: boolean,
   isSelfAssignable: boolean,
-  sectionCodes: string[],
   existingAssignment?: ExistingReviewAssignment
 ): AssignmentState => {
-  const isReviewAssigned = existingReviewsAssigned.length > 0
   const isAssigned = existingReviewsAssigned.some(
     ({ userId }) => userId === existingAssignment?.userId
   )
-  // temporarily final decision shouldn't be locked if there are other reviewAssignment assigned
-  // Note: This logic will be updated during implementation of ISSUE #836 (front-end) to allow
-  // locking other reviewAssignments for finalDecision once one has been submitted.
-  if (canMakeFinalDecision)
-    return {
-      status: ReviewAssignmentStatus.Assigned,
-      isSelfAssignable: true,
-      isLocked: false,
-      assignedSections: sectionCodes,
-    }
-
-  // Create new OR update ReviewAssignment:
+  // Create NEW or update EXISTING ReviewAssignment:
   // 1. If existing
   //   - keep same status, isSelfAssignable
-  //   - just update isLocked = true (if already assigned to another)
+  //   - keep same isLocked if assigned
```

Comment:
> `   //   - keep same isLocked if assigned`

Shouldn't existing review_assignments get `isLocked = false` always? (assuming this action runs on submission of applications and reviews). Otherwise, when do the "locked" assignments get "unlocked"?

Reply:
Fixed. But basically I added one specific case for when one ReviewAssignment was added while in the state of last submitted Review has locked others (LOQ submitted). So for that reason I have also changed `updateReviewStatus` which will set ReviewAssignment to be locked in this case (and then checked here). Please have a look on these 2 files:
- plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts
- plugins/action_update_review_status/src/updateReviewsStatuses.ts